### DISCUSSION TITLE
an undo button

### DISCUSSION THREAD

**Discussion by BinHawkins42** [tsunamayo/Starship-EVO#3967] [Suggestions]
I don't know if this has been suggested in the past but with the implantation of the new prefabs system, I thought now would be a good time to bring it up again. I have had quite a few times in the past few months that an undo button would have been very nice.  I know this is a quality-of-life improvement that does not contribute that much to the game play, but man it's nice when building with larger prefabs that take a while to delete when you get them in the wrong place. 

So, if you could find the time and add that in that would be awesome.

**Comment by cloudrambler** [comment]
What about the other way of looking at the problem. When adding a prefab it could be included as a fixed holographic preview that would allow you to check all the fits before confirming it as a new build. The confirmation could be done with the repair tool for example. Or if you decide it's not what you want, the delete function would remove the whole prefab (preview).

Once it's confirmed, the prefab would convert to normal blocks and then be subject to individual block deletion.

In fact, this method could be suitable for survival style construction too; perhaps only converting one block at a time like the repair tool interaction with damaged blocks.

**Reply by BinHawkins42** [reply]
an interesting way to look at it. but the undo button would be good for more than just prefab construction. it would help with accidental deletion of a block that you didn't want to delete.

**Reply by cloudrambler** [reply]
I agree that an undo function would be generally useful, and it was partially discussed in #3437. My reason for suggesting the alternative method is in case Tsuna has reasons to avoid the undo method, as he did not seem to warm to it when mentioned before.
